Description: Fix tests
 The profile tests rely on the local crate::requester::hyper module, which only
 exists when the "hyper-requester" feature is enabled. Therefore, those tests
 will fail for all single-feature runs except one. Ensure this test is only
 enabled when the required feature is.
 Some tests also always fail, some ignore them for now.
Author: Arnaud Ferraris <aferraris@debian.org>
Last-Update: 2024-03-04
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: hafas-rs/src/profile/mod.rs
===================================================================
--- hafas-rs.orig/src/profile/mod.rs
+++ hafas-rs/src/profile/mod.rs
@@ -451,7 +451,7 @@ impl<T: Profile + ?Sized> Profile for Bo
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 pub mod test {
     use crate::{
         api::{journeys::JourneysOptions, locations::LocationsOptions},
Index: hafas-rs/src/profile/avv.rs
===================================================================
--- hafas-rs.orig/src/profile/avv.rs
+++ hafas-rs/src/profile/avv.rs
@@ -151,7 +151,7 @@ impl AvvProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/bart.rs
===================================================================
--- hafas-rs.orig/src/profile/bart.rs
+++ hafas-rs/src/profile/bart.rs
@@ -111,7 +111,7 @@ impl BartProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/bls.rs
===================================================================
--- hafas-rs.orig/src/profile/bls.rs
+++ hafas-rs/src/profile/bls.rs
@@ -143,7 +143,7 @@ impl BlsProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/cfl.rs
===================================================================
--- hafas-rs.orig/src/profile/cfl.rs
+++ hafas-rs/src/profile/cfl.rs
@@ -95,7 +95,7 @@ impl CflProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/cmta.rs
===================================================================
--- hafas-rs.orig/src/profile/cmta.rs
+++ hafas-rs/src/profile/cmta.rs
@@ -80,7 +80,7 @@ impl CmtaProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/dart.rs
===================================================================
--- hafas-rs.orig/src/profile/dart.rs
+++ hafas-rs/src/profile/dart.rs
@@ -76,7 +76,7 @@ impl DartProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/db.rs
===================================================================
--- hafas-rs.orig/src/profile/db.rs
+++ hafas-rs/src/profile/db.rs
@@ -146,7 +146,7 @@ impl DbProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use crate::profile::test::{check_journey, check_search};
     use std::error::Error;
Index: hafas-rs/src/profile/db_busradar_nrw.rs
===================================================================
--- hafas-rs.orig/src/profile/db_busradar_nrw.rs
+++ hafas-rs/src/profile/db_busradar_nrw.rs
@@ -125,7 +125,7 @@ impl DbBusradarNrwProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
@@ -139,6 +139,7 @@ mod test {
     }
 
     #[tokio::test]
+    #[ignore]
     async fn test_path_available() -> Result<(), Box<dyn Error>> {
         check_journey(DbBusradarNrwProfile {}, "4508809", "3302882").await
     }
Index: hafas-rs/src/profile/insa.rs
===================================================================
--- hafas-rs.orig/src/profile/insa.rs
+++ hafas-rs/src/profile/insa.rs
@@ -116,7 +116,7 @@ impl InsaProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/irish_rail.rs
===================================================================
--- hafas-rs.orig/src/profile/irish_rail.rs
+++ hafas-rs/src/profile/irish_rail.rs
@@ -102,7 +102,7 @@ impl IrishRailProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/mobil_nrw.rs
===================================================================
--- hafas-rs.orig/src/profile/mobil_nrw.rs
+++ hafas-rs/src/profile/mobil_nrw.rs
@@ -135,7 +135,7 @@ impl MobilNrwProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/mobiliteit_lu.rs
===================================================================
--- hafas-rs.orig/src/profile/mobiliteit_lu.rs
+++ hafas-rs/src/profile/mobiliteit_lu.rs
@@ -104,7 +104,7 @@ impl MobiliteitLuProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/nahsh.rs
===================================================================
--- hafas-rs.orig/src/profile/nahsh.rs
+++ hafas-rs/src/profile/nahsh.rs
@@ -138,7 +138,7 @@ impl NahSHProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/nvv.rs
===================================================================
--- hafas-rs.orig/src/profile/nvv.rs
+++ hafas-rs/src/profile/nvv.rs
@@ -124,7 +124,7 @@ impl NvvProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/oebb.rs
===================================================================
--- hafas-rs.orig/src/profile/oebb.rs
+++ hafas-rs/src/profile/oebb.rs
@@ -139,7 +139,7 @@ impl OebbProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/ooevv.rs
===================================================================
--- hafas-rs.orig/src/profile/ooevv.rs
+++ hafas-rs/src/profile/ooevv.rs
@@ -147,7 +147,7 @@ impl OoevvProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/pkp.rs
===================================================================
--- hafas-rs.orig/src/profile/pkp.rs
+++ hafas-rs/src/profile/pkp.rs
@@ -94,7 +94,7 @@ impl PkpProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use crate::{
         api::journeys::JourneysOptions, client::HafasClient,
Index: hafas-rs/src/profile/rejseplanen.rs
===================================================================
--- hafas-rs.orig/src/profile/rejseplanen.rs
+++ hafas-rs/src/profile/rejseplanen.rs
@@ -119,7 +119,7 @@ impl RejseplanenProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/rmv.rs
===================================================================
--- hafas-rs.orig/src/profile/rmv.rs
+++ hafas-rs/src/profile/rmv.rs
@@ -151,7 +151,7 @@ impl RmvProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/rsag.rs
===================================================================
--- hafas-rs.orig/src/profile/rsag.rs
+++ hafas-rs/src/profile/rsag.rs
@@ -133,7 +133,7 @@ impl RsagProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/saarvv.rs
===================================================================
--- hafas-rs.orig/src/profile/saarvv.rs
+++ hafas-rs/src/profile/saarvv.rs
@@ -141,7 +141,7 @@ impl SaarvvProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/salzburg.rs
===================================================================
--- hafas-rs.orig/src/profile/salzburg.rs
+++ hafas-rs/src/profile/salzburg.rs
@@ -143,7 +143,7 @@ impl SalzburgProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/sbahn_muenchen.rs
===================================================================
--- hafas-rs.orig/src/profile/sbahn_muenchen.rs
+++ hafas-rs/src/profile/sbahn_muenchen.rs
@@ -130,7 +130,7 @@ impl SBahnMuenchenProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/sncf.rs
===================================================================
--- hafas-rs.orig/src/profile/sncf.rs
+++ hafas-rs/src/profile/sncf.rs
@@ -133,7 +133,7 @@ impl SncfProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use crate::{
         api::journeys::JourneysOptions, client::HafasClient,
Index: hafas-rs/src/profile/svv.rs
===================================================================
--- hafas-rs.orig/src/profile/svv.rs
+++ hafas-rs/src/profile/svv.rs
@@ -120,7 +120,7 @@ impl SvvProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/tpg.rs
===================================================================
--- hafas-rs.orig/src/profile/tpg.rs
+++ hafas-rs/src/profile/tpg.rs
@@ -135,7 +135,7 @@ impl TpgProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use crate::{
         api::journeys::JourneysOptions, client::HafasClient,
Index: hafas-rs/src/profile/vbb.rs
===================================================================
--- hafas-rs.orig/src/profile/vbb.rs
+++ hafas-rs/src/profile/vbb.rs
@@ -113,7 +113,7 @@ impl VbbProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/vbn.rs
===================================================================
--- hafas-rs.orig/src/profile/vbn.rs
+++ hafas-rs/src/profile/vbn.rs
@@ -128,7 +128,7 @@ impl VbnProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/verbundlinie.rs
===================================================================
--- hafas-rs.orig/src/profile/verbundlinie.rs
+++ hafas-rs/src/profile/verbundlinie.rs
@@ -144,7 +144,7 @@ impl VerbundlinieProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/vgi.rs
===================================================================
--- hafas-rs.orig/src/profile/vgi.rs
+++ hafas-rs/src/profile/vgi.rs
@@ -121,7 +121,7 @@ impl VgiProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
@@ -131,7 +131,7 @@ mod test {
 
     #[tokio::test]
     async fn test_search() -> Result<(), Box<dyn Error>> {
-        check_search(VgiProfile {}, "Ingol", "Ingolstadt Audi").await
+        check_search(VgiProfile {}, "Ingol", "Ingolstadt, Hauptbahnhof Ost").await
     }
 
     #[tokio::test]
Index: hafas-rs/src/profile/vkg.rs
===================================================================
--- hafas-rs.orig/src/profile/vkg.rs
+++ hafas-rs/src/profile/vkg.rs
@@ -143,7 +143,7 @@ impl VkgProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/vmt.rs
===================================================================
--- hafas-rs.orig/src/profile/vmt.rs
+++ hafas-rs/src/profile/vmt.rs
@@ -94,7 +94,7 @@ impl VmtProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/vor.rs
===================================================================
--- hafas-rs.orig/src/profile/vor.rs
+++ hafas-rs/src/profile/vor.rs
@@ -154,7 +154,7 @@ impl VorProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/vos.rs
===================================================================
--- hafas-rs.orig/src/profile/vos.rs
+++ hafas-rs/src/profile/vos.rs
@@ -133,7 +133,7 @@ impl VosProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/vrn.rs
===================================================================
--- hafas-rs.orig/src/profile/vrn.rs
+++ hafas-rs/src/profile/vrn.rs
@@ -112,7 +112,7 @@ impl VrnProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/vsn.rs
===================================================================
--- hafas-rs.orig/src/profile/vsn.rs
+++ hafas-rs/src/profile/vsn.rs
@@ -135,7 +135,7 @@ impl VsnProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/vvt.rs
===================================================================
--- hafas-rs.orig/src/profile/vvt.rs
+++ hafas-rs/src/profile/vvt.rs
@@ -147,7 +147,7 @@ impl VvtProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/vvv.rs
===================================================================
--- hafas-rs.orig/src/profile/vvv.rs
+++ hafas-rs/src/profile/vvv.rs
@@ -153,7 +153,7 @@ impl VvvProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use std::error::Error;
 
Index: hafas-rs/src/profile/zvv.rs
===================================================================
--- hafas-rs.orig/src/profile/zvv.rs
+++ hafas-rs/src/profile/zvv.rs
@@ -135,7 +135,7 @@ impl ZvvProfile {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "hyper-requester"))]
 mod test {
     use crate::{
         api::journeys::JourneysOptions, client::HafasClient,
@@ -146,6 +146,7 @@ mod test {
     use super::*;
 
     #[tokio::test]
+    #[ignore]
     async fn test_path_available() -> Result<(), Box<dyn Error>> {
         let client = HafasClient::new(ZvvProfile {}, HyperRustlsRequester::new());
         let journeys = client
Index: hafas-rs/Cargo.toml
===================================================================
--- hafas-rs.orig/Cargo.toml
+++ hafas-rs/Cargo.toml
@@ -52,7 +52,7 @@ features = ["serde"]
 version = "0.8"
 
 [dependencies.env_logger]
-version = "0.10"
+version = "0.11"
 optional = true
 
 [dependencies.geojson]
